specs/projectreuniondeployment/projectreuniondeployment.md

```diff
@@ -0,0 +1,138 @@
+due to dropoff during the acquistion of Project Reunion.
+
+# Description
+
+When a Project Reunion-consuming package or app is deployed, it is expected to only have the App
+package or installer itself, and the Project Reunion main and framework packages may not be
+present, or may not be at the desired version. This API enables a developer to detect when the
+required Project Reunion main and/or framework packages are present, and to direct the user to the
+appropriate resources to acquire or update Project Reunion if it is absent or requires updates.
+
+Since this API is part of the Project Reunion framework, the framework itself must be present.
+For MSIX packages, the framework must be listed as a dependency in the application manifest and
+will not be deployable without a min version being present. For Win32 programs and installers the
+Project Reunion bootstrapper will try to load the framework and error out with a similar
+instruction on how to resolve the missing dependencies Project Reunion.
+
+Developers must not assume that the Project Reunion main package is present, or that the version
+present is the minimum version required to support the feature set they expect. Instead, this API
+should be used to validate any Project Reunion requirements and direct the user to resolve any
+missing dependencies.
+
+The developer scenarios for this API is to answer the questions:
+1. Does this system have the min version of Project Reunion I require?
+2. What do I do to resolve this if it doesn't?
+
+The API addresses #1 by a simple query of the version they seek and a possible check for OS
+updates. #2 has two possible answers. First is direct resolution by the developer by acquiring and
+installing the packages if possible. Some programs will not have permission or motivation to do
+this, so the second answer is providing a link up to the user where the user can take action to
+install the package. The API will address this scenario by providing an answer as to what is
+required, where to get it, and where to go for instruction.
+
+# Examples
+
+## GetStatus
+
+This is the means by which an app or program can call to verify Project Reunion is present to a
+minimum version. It should be called once after activation before trying to use Project Reunion
+functionality. This method returns a ProjectReunionStatus object that describes the current
+deployment state of Project Reunion.
+
+```C++
+Windows::ApplicationModel::PackageVersion version{1, 0, 0, 0};
+auto status =
+Microsoft::Management::Deployment::ProjectReunionDeployment::GetStatus(version);
+if (!status.IsOK())
+{
+    ...
+}
+```
+
+## GetInstallPackageLink
+
+Provides a direct install link to deploy the latest version of Project Reunion. If no OS Updates
+are required and the program or app has permissions to deploy packages, this can directly resolve
+the missing dependency without burdening the user.
```

Comment:
side note: we should never burden the user

Reply:
I agree, though this is a special circumstance due to permissions. The app may not have permissions to install OS updates or other things required by Project Reunion on a downlevel device. This should be an at most 1 time deal for the typical user, and only on downlevel OSes. For future OSes, this can be mitigated by preinstalling Project Reunion, where it will be auto-updated by the store and required OS updates are already present.